### PR TITLE
Docs Improvement: Groups Overview

### PR DESCRIPTION
**Pull request by gdjohnson**
This branch adds an overview to our new Groups functionality, using as sources the Technical PRD spec and conversations with @zakhap.

## Link to Issue
Closes: #5427

### CODE REVIEW

**Review by rbennettcw — APPROVED** [review]

File: knowledge_base/Groups.md [hunk 1]
```diff
@@ -7,6 +8,16 @@
   * [Update/delete Group](#update-delete-group)
 - [Change Log](#change-log)
 
+# Introduction to Groups
+
+`Groups` are used by community admins to handle permissioning across the forum, using associated `Membership` objects that link `Addresses` to Groups. 
+
+Forum actions are always topic-scoped; whenever a user creates a new thread, reacts to a comment, or participates in a poll, they are operating within the scope of a topic. Groups allow admins to "gate" user actions, within a given topic-scope, based on a set of Requirements.
```
Comment (jnaviask) [inline knowledge_base/Groups.md:15]: Can we add an "e.g. creating a thread or a comment" to "user actions" so we're clear about what kinds of actions we mean?

File: knowledge_base/Groups.md [hunk 2]
```diff
@@ -7,6 +8,16 @@
...
+
+Groups are defined by sets of `Requirements`. A Requirement is a rule or logical statement that qualifies a given address for Group membership. As of 231031, the two Requirements supported on our backend are token thresholds (i.e., that a given address has a certain token balance) and allowlists (i.e., whitelists manually constructed by a community admin). 
+
+From a database perspective, Groups consist of a metadata object, an associated Community (chain), an associated Membership, and an array of Requirements. Requirement objects consist of a `rule` string (e.g. "threshold") and a data object (e.g. `ThresholdData`). Membership validation checks, and requirement types can be found in the `/server/util/requirementsModule` directory.
```
Comment (jnaviask) [inline knowledge_base/Groups.md:19]: We should specify that the requirement object must conform to the provided JSON schema (`packages/commonwealth/server/util/requirementsModule/requirementsSchema_v1.json`).

**Review by jnaviask — CHANGES_REQUESTED** [review]
A few inline comments, + we should also be clear about when memberships are checked and when/if membership objects are created or deleted.

File: knowledge_base/Groups.md (hunk 1 above)
Comment (gdjohnson) [reply knowledge_base/Groups.md]: Edited for clarity!

**Review by jnaviask — APPROVED** [review]

**Pull request merged**